### PR TITLE
feat: add human-readable slugs to group URLs

### LINKED ISSUE #26: groups should have slugs
slugs should be displayed in the url for groups instead of uuids

### PR DESCRIPTION
**Pull request by mwoods79**
## Summary
Replaces UUID-based URLs for groups with human-readable slugs to improve user experience and URL shareability. Groups are now accessible via URLs like `/groups/phoenix-elixir-meetup` instead of `/groups/123e4567-e89b-12d3-a456-426614174000`.

Closes #26

## Changes
- Added Slugify library for robust slug generation with unicode support
- Updated Group resource with slug attribute and auto-generation logic
- Migrated all routes from UUID to slug-based patterns
- Implemented slug editing with URL change warnings
- Added comprehensive unicode support and testing
- Updated seed data to use meaningful group names

## Key Implementation Details
- Slugs are auto-generated from group names using Ash change modules
- Full unicode transliteration support (e.g., "北京用户组" → "bei-jing-yong-hu-zu")
- Unique constraint enforced at database level
- Edit form shows warning when changing slugs
- No backward compatibility (as requested - not in production)

## Testing
- Run the app: `mix phx.server`
- Create a group and see slug auto-generation
- Try unicode names like "Café München" or "北京用户组"
- Edit a group and change the slug (see warning)
- Navigate using new slug-based URLs

### Test Coverage
- 212 tests passing (added 3 new unicode tests)
- All Cucumber feature tests passing
- Comprehensive slug generation and validation tests
- Edge cases covered (empty slugs, collisions, unicode)

## Learnings
1. **Ash Migration Integrity**: Never manually edit Ash-generated migrations - it breaks snapshot tracking. Always regenerate for clean implementation.
2. **Unicode Excellence**: The Slugify library provides excellent international support through transliteration, making the feature globally accessible.

## Screenshots
Example slugs generated:
- "Phoenix Elixir Meetup" → `/groups/phoenix-elixir-meetup`
- "Café München" → `/groups/cafe-munchen` 
- "北京用户组" → `/groups/bei-jing-yong-hu-zu`

### CODE REVIEW

File: lib/huddlz_web/live/group_live/show.ex [hunk 1]
```diff
@@ -217,8 +223,12 @@ defmodule HuddlzWeb.GroupLive.Show do
     end
   end
 
-  defp get_group(id) do
-    case Ash.get(Group, id, load: [:owner], authorize?: false) do
+  defp get_group_by_slug(slug) do
+    case Group
+         |> Ash.Query.for_read(:get_by_slug, %{slug: slug})
+         |> Ash.Query.load(:owner)
+         |> Ash.read_one(authorize?: false) do
```
Comment (mwoods79) [inline lib/huddlz_web/live/group_live/show.ex:230]: should not `authorize: false` in any LiveView code, really should only do it in tests, but there are places where exceptions can be made.

File: lib/huddlz_web/live/group_live/show.ex [hunk 2]
```diff
@@ -325,7 +335,7 @@ defmodule HuddlzWeb.GroupLive.Show do
     |> Ash.Query.filter(group_id == ^group.id)
     |> Ash.Query.filter(starts_at > ^DateTime.utc_now())
     |> Ash.Query.sort(starts_at: :asc)
-    |> Ash.Query.load([:status, :visible_virtual_link])
+    |> Ash.Query.load([:status, :visible_virtual_link, :group])
     |> Ash.read!(actor: user)
```
Comment (mwoods79) [inline lib/huddlz_web/live/group_live/show.ex:339]: This shoudl probably be on the domain (and tested)

File: lib/huddlz_web/live/group_live/edit.ex [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+        {:noreply, assign(socket, :form, form)}
+    end
+  end
+
+  defp get_group_by_slug(slug) do
+    case Group
+         |> Ash.Query.for_read(:get_by_slug, %{slug: slug})
+         |> Ash.Query.load(:owner)
+         |> Ash.read_one(authorize?: false) do
```
Comment (mwoods79) [inline lib/huddlz_web/live/group_live/edit.ex:189]: should probably be called from domain

File: lib/huddlz/communities/group.ex [hunk 4]
```diff
@@ -44,9 +45,20 @@ defmodule Huddlz.Communities.Group do
       filter expr(owner_id == ^arg(:owner_id))
     end
 
+    read :get_by_slug do
```
Comment (mwoods79) [inline lib/huddlz/communities/group.ex:48]: needs to be added to policies and a resource test for the policy added.
